plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1782,8 +1783,8 @@ private ProtocolEntry getProtocolEntry(Map<String, Object> properties)
             }
         }
         return new ProtocolEntry(
-                readerVersion.orElse(MIN_READER_VERSION),
-                writerVersion.orElse(MIN_WRITER_VERSION));
+                readerVersion.orElse(deltaLakeConfig.getReaderVersion()),
+                writerVersion.orElse(deltaLakeConfig.getWriterVersion()));
```

Comment:
I couldn't understand the intention of this change. Did you want to set default reader and writer versions of `DeltaLakeConfig`?

Reply:
Yes I want to set default reader and writer version.

Reply:
Why do you initialize `DeltaLakeConfig` at L1767? It ignores configured values by users.

Reply:
My intension was, If `reader_version` and `writer_version` are overridden from `CREATE TABLE ...` or `ALTER TABLE ...` statement then use it, otherwise use it from the delta config properties.

I got the problem over here. I have to use configured DeltaLakeConfig and not with default values only.

Reply:
I see that we get `DeltaLakeConfig` properties in `DeltaLakeMetadataFactory` and pass them to `DeltaLakeMetadata` as arguments. This way it will keep on increasing the number of arguments if we want to use more and more `DeltaLakeConfig` properties in `DeltaLakeMetadata`.

https://github.com/trinodb/trino/blob/c12f3a69e3eed14ccff715fbaa6d9fcf6b471ef3/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadataFactory.java#L97-L102

WDYT, If we refactor (in a separate commit) DeltaLakeMetadataFactory to pass `DeltaLakeConfig` as an argument to  `DeltaLakeMetadata`? In growing use of `DeltaLakeConfig` properties in `DeltaLakeMetadata` will ensure no change in `DeltaLakeMetadataFactory`.